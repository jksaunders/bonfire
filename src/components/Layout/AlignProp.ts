import { LayoutProps } from './Layout';

export type AlignCombinations =
  | '_'
  | 'center'
  | 'start'
  | 'flex-end'
  | 'flex-start'
  | 'end'
  | 'content-center'
  | 'content-start'
  | 'content-flex-end'
  | 'content-flex-start'
  | 'content-end'
  | 'content-space-between'
  | 'content-space-around'
  | 'content-space-evenly'
  | 'content-stretch'
  | 'center _'
  | 'center center'
  | 'center start'
  | 'center flex-end'
  | 'center flex-start'
  | 'center end'
  | 'center content-center'
  | 'center content-start'
  | 'center content-flex-end'
  | 'center content-flex-start'
  | 'center content-end'
  | 'center content-space-between'
  | 'center content-space-around'
  | 'center content-space-evenly'
  | 'center content-stretch'
  | 'start _'
  | 'start center'
  | 'start start'
  | 'start flex-end'
  | 'start flex-start'
  | 'start end'
  | 'start content-center'
  | 'start content-start'
  | 'start content-flex-end'
  | 'start content-flex-start'
  | 'start content-end'
  | 'start content-space-between'
  | 'start content-space-around'
  | 'start content-space-evenly'
  | 'start content-stretch'
  | 'flex-end _'
  | 'flex-end center'
  | 'flex-end start'
  | 'flex-end flex-end'
  | 'flex-end flex-start'
  | 'flex-end end'
  | 'flex-end content-center'
  | 'flex-end content-start'
  | 'flex-end content-flex-end'
  | 'flex-end content-flex-start'
  | 'flex-end content-end'
  | 'flex-end content-space-between'
  | 'flex-end content-space-around'
  | 'flex-end content-space-evenly'
  | 'flex-end content-stretch'
  | 'flex-start _'
  | 'flex-start center'
  | 'flex-start start'
  | 'flex-start flex-end'
  | 'flex-start flex-start'
  | 'flex-start end'
  | 'flex-start content-center'
  | 'flex-start content-start'
  | 'flex-start content-flex-end'
  | 'flex-start content-flex-start'
  | 'flex-start content-end'
  | 'flex-start content-space-between'
  | 'flex-start content-space-around'
  | 'flex-start content-space-evenly'
  | 'flex-start content-stretch'
  | 'end _'
  | 'end center'
  | 'end start'
  | 'end flex-end'
  | 'end flex-start'
  | 'end end'
  | 'end content-center'
  | 'end content-start'
  | 'end content-flex-end'
  | 'end content-flex-start'
  | 'end content-end'
  | 'end content-space-between'
  | 'end content-space-around'
  | 'end content-space-evenly'
  | 'end content-stretch'
  | 'content-center _'
  | 'content-center center'
  | 'content-center start'
  | 'content-center flex-end'
  | 'content-center flex-start'
  | 'content-center end'
  | 'content-center content-center'
  | 'content-center content-start'
  | 'content-center content-flex-end'
  | 'content-center content-flex-start'
  | 'content-center content-end'
  | 'content-center content-space-between'
  | 'content-center content-space-around'
  | 'content-center content-space-evenly'
  | 'content-center content-stretch'
  | 'content-start _'
  | 'content-start center'
  | 'content-start start'
  | 'content-start flex-end'
  | 'content-start flex-start'
  | 'content-start end'
  | 'content-start content-center'
  | 'content-start content-start'
  | 'content-start content-flex-end'
  | 'content-start content-flex-start'
  | 'content-start content-end'
  | 'content-start content-space-between'
  | 'content-start content-space-around'
  | 'content-start content-space-evenly'
  | 'content-start content-stretch'
  | 'content-flex-end _'
  | 'content-flex-end center'
  | 'content-flex-end start'
  | 'content-flex-end flex-end'
  | 'content-flex-end flex-start'
  | 'content-flex-end end'
  | 'content-flex-end content-center'
  | 'content-flex-end content-start'
  | 'content-flex-end content-flex-end'
  | 'content-flex-end content-flex-start'
  | 'content-flex-end content-end'
  | 'content-flex-end content-space-between'
  | 'content-flex-end content-space-around'
  | 'content-flex-end content-space-evenly'
  | 'content-flex-end content-stretch'
  | 'content-flex-start _'
  | 'content-flex-start center'
  | 'content-flex-start start'
  | 'content-flex-start flex-end'
  | 'content-flex-start flex-start'
  | 'content-flex-start end'
  | 'content-flex-start content-center'
  | 'content-flex-start content-start'
  | 'content-flex-start content-flex-end'
  | 'content-flex-start content-flex-start'
  | 'content-flex-start content-end'
  | 'content-flex-start content-space-between'
  | 'content-flex-start content-space-around'
  | 'content-flex-start content-space-evenly'
  | 'content-flex-start content-stretch'
  | 'content-end _'
  | 'content-end center'
  | 'content-end start'
  | 'content-end flex-end'
  | 'content-end flex-start'
  | 'content-end end'
  | 'content-end content-center'
  | 'content-end content-start'
  | 'content-end content-flex-end'
  | 'content-end content-flex-start'
  | 'content-end content-end'
  | 'content-end content-space-between'
  | 'content-end content-space-around'
  | 'content-end content-space-evenly'
  | 'content-end content-stretch'
  | 'content-space-between _'
  | 'content-space-between center'
  | 'content-space-between start'
  | 'content-space-between flex-end'
  | 'content-space-between flex-start'
  | 'content-space-between end'
  | 'content-space-between content-center'
  | 'content-space-between content-start'
  | 'content-space-between content-flex-end'
  | 'content-space-between content-flex-start'
  | 'content-space-between content-end'
  | 'content-space-between content-space-between'
  | 'content-space-between content-space-around'
  | 'content-space-between content-space-evenly'
  | 'content-space-between content-stretch'
  | 'content-space-around _'
  | 'content-space-around center'
  | 'content-space-around start'
  | 'content-space-around flex-end'
  | 'content-space-around flex-start'
  | 'content-space-around end'
  | 'content-space-around content-center'
  | 'content-space-around content-start'
  | 'content-space-around content-flex-end'
  | 'content-space-around content-flex-start'
  | 'content-space-around content-end'
  | 'content-space-around content-space-between'
  | 'content-space-around content-space-around'
  | 'content-space-around content-space-evenly'
  | 'content-space-around content-stretch'
  | 'content-space-evenly _'
  | 'content-space-evenly center'
  | 'content-space-evenly start'
  | 'content-space-evenly flex-end'
  | 'content-space-evenly flex-start'
  | 'content-space-evenly end'
  | 'content-space-evenly content-center'
  | 'content-space-evenly content-start'
  | 'content-space-evenly content-flex-end'
  | 'content-space-evenly content-flex-start'
  | 'content-space-evenly content-end'
  | 'content-space-evenly content-space-between'
  | 'content-space-evenly content-space-around'
  | 'content-space-evenly content-space-evenly'
  | 'content-space-evenly content-stretch'
  | 'content-stretch _'
  | 'content-stretch center'
  | 'content-stretch start'
  | 'content-stretch flex-end'
  | 'content-stretch flex-start'
  | 'content-stretch end'
  | 'content-stretch content-center'
  | 'content-stretch content-start'
  | 'content-stretch content-flex-end'
  | 'content-stretch content-flex-start'
  | 'content-stretch content-end'
  | 'content-stretch content-space-between'
  | 'content-stretch content-space-around'
  | 'content-stretch content-space-evenly'
  | 'content-stretch content-stretch';

const parseAlignTargetValue = (
  alignProp: AlignCombinations
): { contentOrItemsKey: 'content' | 'items'; value: string } => ({
  contentOrItemsKey: alignProp.includes('content-') ? 'content' : 'items',
  value: `${alignProp}`.replace('items-', '').replace('content-', ''),
});

export const getAlignmentCss = (alignmentProp: AlignCombinations): string => {
  const [first, second] = alignmentProp.split(' ') as AlignCombinations[];
  let justifyResult = '';
  let alignResult = '';

  if (first !== '_') {
    const {
      contentOrItemsKey: firstContentOrItemsKey,
      value: firstValue,
    } = parseAlignTargetValue(first);
    justifyResult = `${firstContentOrItemsKey}: ${firstValue}`;
  }

  if (second == null) {
    alignResult = justifyResult;
  } else if (second !== '_') {
    const {
      contentOrItemsKey: secondContentOrItemsKey,
      value: secondValue,
    } = parseAlignTargetValue(second);
    alignResult = `${secondContentOrItemsKey}: ${secondValue}`;
  }

  return `
    ${alignResult && `align-${alignResult};`}
    ${justifyResult && `justify-${justifyResult};`}
  `;
};
