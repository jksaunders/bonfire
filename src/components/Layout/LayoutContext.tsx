import { createContext } from 'react';

export interface LayoutContextProps {
  defaultGap?: string;
}

const LayoutContext = createContext<LayoutContextProps>({});

export default LayoutContext;
