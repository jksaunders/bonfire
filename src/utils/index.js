export { default as Browser } from "./browser";
export * from "./styling";