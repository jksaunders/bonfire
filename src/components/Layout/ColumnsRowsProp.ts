import { sizesObjectToCss } from '../../utils/CssRule/CssRule';
import { LayoutProps } from './Layout';

export const getColumnsRowsCss = (direction: 'columns' | 'rows') => (
  props: LayoutProps
): string => {
  return sizesObjectToCss(props[direction], (value) => {
    const key =
      value.includes(' ') || value.includes('repeat')
        ? `grid-template-${direction}`
        : `grid-auto-${direction}`;
    return `${key}: ${value};`;
  });
};
