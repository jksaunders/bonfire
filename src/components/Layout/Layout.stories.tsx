import React from 'react';
import styled from 'styled-components';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { Layout, LayoutProps } from './Layout';

export default {
  title: 'Components/Layout',
  component: Layout,
  argTypes: {
    align: { control: 'text' },
    column: { control: 'boolean' },
    full: { control: 'boolean' },
    fullHeight: { control: 'boolean' },
    fullWidth: { control: 'boolean' },
    gap: { control: 'text' },
    row: { control: 'boolean' },
  },
} as Meta;

const Content = styled.div`
  background-color: lightblue;
  padding: 10px;
`;

const Template: Story<LayoutProps> = (args) => (
  <Layout {...args}>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
    <Content>hi</Content>
  </Layout>
);

export const Primary = Template.bind({});
Primary.args = {};
