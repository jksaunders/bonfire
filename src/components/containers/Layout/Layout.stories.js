import React from "react";
import styled from "styled-components";

import { storiesOf } from "@storybook/react";
import {
  boolean,
  text,
  withKnobs
} from "@storybook/addon-knobs";

import Layout from "./Layout";
import Typography, { MaterialVariants, TrelloVariants } from "../../Typography";

const StoryWrapper = styled.div`
  background-color: lightblue;
  height: ${props => props.size};
  width: ${props => props.size};
`;

const Content = styled.div`
  background-color: rgba(255, 255, 255, 0.7);
`;

storiesOf("Layout", module)
  .addDecorator(withKnobs)
  .add("basic", () => {
    const centered = boolean("Centered?", false);
    const containerSize = text("Size of container", "500px");
    const columns = text("Columns", "");
    const gap = text("Gap", "");
    const horizontalAlignment = text("Horizontal alignment", "");
    const rows = text("Rows", "");
    const padding = text("Padding", "");
    const verticalAlignment = text("Vertical alignment", "");

    return (
      <StoryWrapper size={containerSize}>
        <Layout
          centered={centered}
          rows={rows !== "" ? rows : null}
          columns={columns !== "" ? columns : null}
          gap={gap !== "" ? gap : null}
          horizontalAlignment={horizontalAlignment !== "" ? horizontalAlignment : null}
          padding={padding !== "" ? padding : null}
          verticalAlignment={verticalAlignment !== "" ? verticalAlignment : null}
        >
          <Content><Typography variant={MaterialVariants.H2}>content1</Typography></Content>
          <Content><Typography variant={MaterialVariants.H2}>content2</Typography></Content>
          <Content><Typography variant={MaterialVariants.H2}>content3</Typography></Content>
        </Layout>
      </StoryWrapper>
    );
  })
  .add("Trello", () => (
    <Layout
      background="#0079bf"
      height="100vh"
      gap="5px"
      rows="max-content max-content auto"
      width="100vw"
    >
      <Layout columns="max-content auto max-content" padding="5px">
        <Layout columns="max-content max-content minmax(max-content, 200px)" gap="5px">
          { ["home", "boards", "search"].map(t => (
            <Layout key={t} background="hsla(0, 0%, 100%, 0.3)" borderRadius="3px" padding="5px">
              <Typography variant={TrelloVariants.Body}>
                {t}
              </Typography>
            </Layout>
          ))}
        </Layout>
        <Layout centered>logo</Layout>
        <Layout>right buttons</Layout>
      </Layout>
      <Layout>
        <Typography variant={TrelloVariants.Body}>
          board header
        </Typography>
      </Layout>
      <Layout columns="272px" flow="column" gap="10px" padding="10px" overflowX="auto">
        { ["To-Do", "Projects", "Bookmarks", "Recipes", "To-Do 2", "Projects 2", "Bookmarks 2", "Recipes 2"].map(t => (
          <Layout key={t} background="#ebecf0" borderRadius="3px" gap="10px" padding="10px" rows="max-content">
            <Layout padding="10px">
              <Typography variant={TrelloVariants.Body}>
                {t}
              </Typography>
            </Layout>
            { [1, 2, 3, 4, 5].map(n => (
              <Layout key={n} background="#FFFFFF" padding="10px">
                <Typography variant={TrelloVariants.Body}>
                  To do
                  {n}
                </Typography>
              </Layout>
            ))}
          </Layout>
        ))}
      </Layout>
    </Layout>
  ));