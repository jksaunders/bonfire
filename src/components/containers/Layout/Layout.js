import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { css, cssBackground, CssRuleType } from '../../../utils/styling';
import { TypographyContext, cssTypography } from '../../Typography';

const propTypes = {
  align: CssRuleType(PropTypes.string),
  background: CssRuleType(PropTypes.string),
  borderRadius: CssRuleType(PropTypes.string),
  centered: PropTypes.bool,
  children: PropTypes.node,
  className: CssRuleType(PropTypes.string),
  columns: CssRuleType(PropTypes.string),
  flow: CssRuleType(PropTypes.string),
  full: PropTypes.bool,
  fullHeight: PropTypes.bool,
  fullWidth: PropTypes.bool,
  gap: CssRuleType(PropTypes.string),
  height: CssRuleType(PropTypes.string),
  horizontalAlignment: CssRuleType(PropTypes.string),
  margin: CssRuleType(PropTypes.string),
  overflow: CssRuleType(PropTypes.string),
  overflowX: CssRuleType(PropTypes.string),
  overflowY: CssRuleType(PropTypes.string),
  rows: CssRuleType(PropTypes.string),
  padding: CssRuleType(PropTypes.string),
  useTypography: PropTypes.bool,
  verticalAlignment: CssRuleType(PropTypes.string),
  width: CssRuleType(PropTypes.string)
};

const defaultProps = {
  align: null,
  background: null,
  borderRadius: null,
  children: null,
  centered: false,
  className: null,
  columns: null,
  flow: null,
  full: false,
  fullHeight: false,
  fullWidth: false,
  gap: null,
  height: null,
  horizontalAlignment: null,
  margin: null,
  overflow: null,
  overflowX: null,
  overflowY: null,
  rows: null,
  padding: null,
  useTypography: null,
  verticalAlignment: null,
  width: null
};

const parseAlignTargetValue = alignProp => ({
  target: alignProp.includes('content-') ? 'content' : 'items',
  value: `${alignProp}`.replace('items-', '').replace('content-', '')
});

const alignment = props => {
  if (props.align == null) {
    return '';
  }

  const [first, second] = props.align.split(' ');
  let justifyResult = '';
  let alignResult = '';

  if (first !== '_') {
    const { target: firstTarget, value: firstValue } = parseAlignTargetValue(first);
    justifyResult = `${firstTarget}: ${firstValue}`;
  }

  if (second == null) {
    alignResult = justifyResult;
  } else if (second !== '_') {
    const { target: secondTarget, value: secondValue } = parseAlignTargetValue(second);
    alignResult = `${secondTarget}: ${secondValue}`;
  }

  return `
    ${alignResult && `align-${alignResult};`}
    ${justifyResult && `justify-${justifyResult};`}
  `;
};

const gridDirectionKey = (direction) => props => {
  if (props[direction] == null) {
    return '';
  }
  if (props[direction].includes(' ') || props[direction].includes('repeat')) {
    return `grid-template-${direction}`;
  }
  return `grid-auto-${direction}`;
};

const Grid = styled.div`
  display: grid;
  box-sizing: border-box;
  ${cssBackground}
  ${css('borderRadius', 'border-radius')}

  ${/* grid properties */''}
  ${css('columns', gridDirectionKey('columns'))}
  ${css('rows', gridDirectionKey('rows'))}
  ${css('gap', 'grid-gap')}
  ${css('flow', 'grid-auto-flow')}

  ${/* alignment */''}
  ${css('centered', 'justify-items', 'center')}
  ${css('centered', 'align-items', 'center')}
  ${alignment}

  ${/* sizing */''}
  ${css(
    ['full', 'height', '100%'],
    ['fullHeight', 'height', '100%'],
    ['height']
  )}
  ${css(
    ['full', 'width', '100%'],
    ['fullWidth', 'width', '100%'],
    ['width']
  )}
  ${css('margin')}
  ${css('padding')}


  ${/* overflow */''}
  ${css('overflow')}
  ${css('overflowX', 'overflow-x')}
  ${css('overflowY', 'overflow-y')}
  
  ${/* typography */''}
  ${cssTypography('typographyProps')}
`;

const Layout = ({
  align,
  background,
  borderRadius,
  centered,
  children,
  className,
  columns,
  flow,
  full,
  fullHeight,
  fullWidth,
  gap,
  height,
  horizontalAlignment,
  margin,
  overflow,
  overflowX,
  overflowY,
  rows,
  padding,
  useTypography,
  verticalAlignment,
  width,
}) => {
  const typographyProps = useContext(TypographyContext);

  return (
    <Grid
      align={align}
      background={background}
      borderRadius={borderRadius}
      centered={centered}
      columns={columns}
      className={className}
      flow={flow}
      full={full}
      fullHeight={fullHeight}
      fullWidth={fullWidth}
      gap={gap}
      height={height}
      horizontalAlignment={horizontalAlignment}
      margin={margin}
      overflow={overflow}
      overflowX={overflowX}
      overflowY={overflowY}
      rows={rows}
      padding={padding}
      verticalAlignment={verticalAlignment}
      width={width}
      typographyProps={useTypography ? typographyProps : {}}
    >
      {children}
    </Grid>
  );
};

Layout.propTypes = propTypes;
Layout.defaultProps = defaultProps;
Layout.shape = propTypes;

export default Layout;