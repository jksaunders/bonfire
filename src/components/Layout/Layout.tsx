import React, { FC, useContext } from 'react';
import styled from 'styled-components';
import { cssRule, sizesObjectToCss } from '../../utils/CssRule/CssRule';
import { ResponsiveProps } from '../../utils/ResponsiveProps';
import { AlignCombinations, getAlignmentCss } from './AlignProp';
import { getColumnsRowsCss } from './ColumnsRowsProp';
import LayoutContext from './LayoutContext';

export interface BaseLayoutProps {
  align?: AlignCombinations;
  column?: boolean;
  columns?: string;
  full?: boolean;
  fullHeight?: boolean;
  fullWidth?: boolean;
  gap?: boolean | string;
  padding?: string;
  row?: boolean;
  rows?: string;
}

export type LayoutProps = ResponsiveProps<BaseLayoutProps>;

const StyledLayout = styled.div<LayoutProps>`
  display: grid;
  ${cssRule<LayoutProps>('padding')}

  ${/* align */ ''}
  ${getAlignmentCss}
  
  ${/* size */ ''}
  ${cssRule<LayoutProps>('full', 'height', '100%')}
  ${cssRule<LayoutProps>('full', 'width', '100%')}
  ${cssRule<LayoutProps>('fullHeight', 'height', '100%')}
  ${cssRule<LayoutProps>('fullWidth', 'width', '100%')}
  
  ${/* grid */ ''}
  ${cssRule<LayoutProps>('gap', 'grid-gap')}
  ${cssRule<LayoutProps>(
    ['column', 'grid-auto-flow', 'row'],
    ['row', 'grid-auto-flow', 'column']
  )}
  ${getColumnsRowsCss('columns')}
  ${getColumnsRowsCss('rows')}
`;

export const Layout: FC<LayoutProps> = ({
  children,
  gap: gapProp,
  ...rest
}) => {
  const layoutContext = useContext(LayoutContext);

  const gap = gapProp === true ? layoutContext.defaultGap : gapProp;

  return (
    <StyledLayout gap={gap} {...rest}>
      {children}
    </StyledLayout>
  );
};
