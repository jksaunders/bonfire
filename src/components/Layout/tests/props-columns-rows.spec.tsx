import React from 'react';
import render from '../../../test-utils/render';
import { ResponsiveProp } from '../../../utils/ResponsiveProps';
import { Layout } from '../Layout';

type StringMediaPair = [value: string, media?: string];

describe('non-responsive', () => {
  interface TestCase {
    columns?: ResponsiveProp<string>;
    expectedGridAutoColumns?: StringMediaPair[];
    expectedGridAutoRows?: StringMediaPair[];
    expectedGridTemplateColumns?: StringMediaPair[];
    expectedGridTemplateRows?: StringMediaPair[];
    rows?: ResponsiveProp<string>;
    title: string;
  }

  const testCases: TestCase[] = [
    {
      title: 'no columns, no rows',
    },
    {
      columns: '10px',
      expectedGridAutoColumns: [['10px']],
      title: 'single value column',
    },
    {
      expectedGridAutoRows: [['10px']],
      rows: '10px',
      title: 'single value rows',
    },
    {
      columns: '10px 12px',
      expectedGridTemplateColumns: [['10px 12px']],
      title: 'multiple value columns',
    },
    {
      expectedGridTemplateRows: [['10px 12px']],
      rows: '10px 12px',
      title: 'multiple value rows',
    },
    {
      // Responsive tests can't check @media style rules
      columns: {
        '_-200px': '10px',
        '200px-_': 'repeat(auto-fill, 10px)',
      },
      expectedGridAutoColumns: [['10px', '@media all and (max-width:200px)']],
      title: 'responsive columns',
    },
  ];

  testCases.forEach(
    ({
      columns,
      expectedGridAutoColumns,
      expectedGridAutoRows,
      expectedGridTemplateColumns,
      expectedGridTemplateRows,
      rows,
      title,
    }) => {
      test(title, () => {
        const { hasStyle, takeSnapshot } = render(
          <Layout columns={columns} rows={rows}>
            content
          </Layout>
        );

        expectedGridAutoColumns?.forEach(([value, media]) => {
          if (media) {
          } else {
            hasStyle(
              { 'grid-auto-columns': value },
              media ? { media } : undefined
            );
          }
        });

        expectedGridAutoRows?.forEach(([value, media]) => {
          if (media) {
          } else {
            hasStyle(
              { 'grid-auto-rows': value },
              media ? { media } : undefined
            );
          }
        });

        expectedGridTemplateColumns?.forEach(([value, media]) => {
          if (media) {
          } else {
            hasStyle(
              { 'grid-template-columns': value },
              media ? { media } : undefined
            );
          }
        });

        expectedGridTemplateRows?.forEach(([value, media]) => {
          if (media) {
          } else {
            hasStyle(
              { 'grid-template-rows': value },
              media ? { media } : undefined
            );
          }
        });

        takeSnapshot();
      });
    }
  );
});
