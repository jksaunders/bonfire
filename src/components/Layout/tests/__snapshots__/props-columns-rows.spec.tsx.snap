// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`non-responsive multiple value columns 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 10px 12px;
}

<div
  class="c0"
>
  content
</div>
`;

exports[`non-responsive multiple value rows 1`] = `
.c0 {
  display: grid;
  grid-template-rows: 10px 12px;
}

<div
  class="c0"
>
  content
</div>
`;

exports[`non-responsive no columns, no rows 1`] = `
.c0 {
  display: grid;
}

<div
  class="c0"
>
  content
</div>
`;

exports[`non-responsive responsive columns 1`] = `
.c0 {
  display: grid;
}

@media all and (max-width:200px) {
  .c0 {
    grid-auto-columns: 10px;
  }
}

@media all and (min-width:200px) {
  .c0 {
    grid-template-columns: repeat(auto-fill,10px);
  }
}

<div
  class="c0"
>
  content
</div>
`;

exports[`non-responsive single value column 1`] = `
.c0 {
  display: grid;
  grid-auto-columns: 10px;
}

<div
  class="c0"
>
  content
</div>
`;

exports[`non-responsive single value rows 1`] = `
.c0 {
  display: grid;
  grid-auto-rows: 10px;
}

<div
  class="c0"
>
  content
</div>
`;
